.jenkins/tests.Jenkinsfile

```diff
@@ -0,0 +1,185 @@
+/* A Jenkins pipeline that will handle PR and nightly tests
+*  These are the original pipelines:
+*  https://github.com/deislabs/mystikos/blob/main/.azure_pipelines/ci-pipeline-makefile.yml
+*  https://github.com/deislabs/mystikos/blob/main/.azure_pipelines/ci-pipeline-makefile-nightly.yml
+*/
+
+pipeline {
+    agent {
+        label 'ACC-1804-DC4'
+    }
+    options {
+        timeout(time: 600, unit: 'MINUTES')
+    }
+    parameters {
+        string(name: "BRANCH_NAME", defaultValue: "main", description: "Branch to build")
+        string(name: "SCRIPTS_ROOT", defaultValue: '${WORKSPACE}/.azure_pipelines/scripts', description: "Root directory")
+        choice(name: "MYST_NIGHTLY_TEST", choices:['0','1'], description: "Enable nightly test")
+    }
+    stages {
+        stage("Build") {
+            when {
+                /* Jobs must meet any of the situations below in order to build:
+                    1. Is started manually, or by a scheduler
+                    2. Is testing a PR to main that contains more than just documentation changes
+                    3. Has MYST_NIGHTLY_TEST set to "1"
+                */
+                anyOf {
+                    triggeredBy 'UserIdCause'
+                    triggeredBy 'TimerTrigger'
+                    expression { params.MYST_NIGHTLY_TEST == "1" }
+                    allOf {
+                        anyOf {
+                            changeRequest target: 'main'
+                            branch 'staging'
+                            branch 'trying'
+                        }
+                        not {
+                            anyOf {
+                                changeset pattern: "doc/*"
+                                changeset pattern: "*(.txt|md)\$", comparator: "REGEXP"
+                            }
+                        }
+                    }
+                }
+            }
+            stages {
+                stage("Cleanup files") {
+                    steps {
+                        sh 'sudo rm -rf /tmp/myst*'
+                        sh 'df'
+                    }
+                }
+                stage('Minimum init config') {
+                    steps {
+                        checkout([$class: 'GitSCM',
+                            branches: [[name: BRANCH_NAME]],
+                            extensions: [],
+                            userRemoteConfigs: [[url: 'https://github.com/cyandevs/mystikos']]])
```

Comment:
Hey, is this intentionally pointing to a fork?

Reply:
Yeah I left this in since tests depend on the UID addition and I've been running these tests on Jenkins. Would you be fine with me leaving this in until this is deemed ready to merge?

Reply:
All tests complete and ready to merge now :)